src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-form/configuration-form.component.ts

```diff
@@ -0,0 +1,267 @@
+      return [this.ipv4Ipv6Validator(configOption)];
+    } else if (configOption.type === 'uuid_d') {
+      return [this.uuidValidator(configOption)];
+    }
+  }
+
+  getStep(type: string, value: number): number | undefined {
+    const numberTypes = ['uint64_t', 'int64_t', 'size_t', 'secs'];
+
+    if (numberTypes.includes(type)) {
+      return 1;
+    }
+
+    if (type === 'double') {
+      if (value !== null) {
+        const stringVal = value.toString();
+        if (stringVal.indexOf('.') !== -1) {
+          // Value type double and contains decimal characters
+          const decimal = value.toString().split('.');
+          return Math.pow(10, -decimal[1].length);
+        }
+      }
+
+      return 0.1;
+    }
+
+    return undefined;
+  }
+
+  private numberValidator(configOption): ValidatorFn[] {
+    const validators = [];
+    if (configOption.max && configOption.max !== '') {
+      this.maxValue = configOption.max;
+      validators.push(Validators.max(configOption.max));
+    }
+    if (configOption.min && configOption.min !== '') {
+      this.minValue = configOption.min;
+      validators.push(Validators.min(configOption.min));
+    }
+    if (configOption.type === 'double') {
+      validators.push(Validators.pattern('^-?[0-9]+(.[0-9]+)?$'));
+    } else {
+      validators.push(Validators.pattern('^-?[0-9]+$'));
+    }
+
+    return validators;
+  }
+
+  private ipv4Ipv6Validator(configOption): ValidatorFn {
+    const ipv4Ipv6Rgx =
+      '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]).){3}([0-9]|[1-9][0-9]|1[0-9]{2}|' +
+      '2[0-4][0-9]|25[0-5])$|^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]).)*([A-Za-z]|[A-Za-z]' +
+      '[A-Za-z0-9-]*[A-Za-z0-9])$|^s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|((' +
+      '[0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d' +
+      '|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|' +
+      '2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}' +
+      '(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+      '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4})' +
+      '{1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|' +
+      '1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|' +
+      '((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d))' +
+      '{3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:' +
+      '((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:' +
+      '[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+      '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*$';
+    return Validators.pattern(ipv4Ipv6Rgx);
+  }
+
+  private uuidValidator(configOption): ValidatorFn {
```

Comment:
Consider relocating this to shared/forms/cd-validators.ts. Think time will come that other forms will use this.

Reply:
I don't think that external contributors will find this validator in this case.

I think we should always locate potential reusable validators to cd-validators from the beginning, otherwise contributors that want to reuse a validator that is located somewhere in the code must relocate the code in their PR which is not good because then code that has nothing to do with the PR must be changed.